src/components/autocomplete/js/autocompleteController.js

```diff
@@ -466,9 +466,9 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        event.preventDefault(); // Prevent browser from always clearing input
```

Comment:
When I first saw this, I assumed that it would always prevent the clearing of the autocomplete, but that doesn't seem to be the case when I actually tested it. Under what conditions does this prevent clearing?